src/operations/distinct.ts

```diff
@@ -73,17 +69,20 @@ export class DistinctOperation extends CommandCallbackOperation<any[]> {
     try {
       decorateWithCollation(cmd, coll, options);
     } catch (err) {
-      return callback(err);
+      return err;
     }
 
```

Comment:
we can remove the try-catch here, because `async` functions automatically wrap any errors that are thrown in promise rejections.